server/src/main/java/io/seata/server/storage/db/lock/DataBaseLockManager.java

```diff
@@ -44,11 +61,19 @@ public boolean releaseLock(BranchSession branchSession) throws TransactionExcept
 
     @Override
     public boolean releaseGlobalSessionLock(GlobalSession globalSession) throws TransactionException {
-        ArrayList<BranchSession> branchSessions = globalSession.getBranchSessions();
+        List<BranchSession> branchSessions = globalSession.getBranchSessions();
         if (CollectionUtils.isEmpty(branchSessions)) {
             return true;
         }
-        List<Long> branchIds = branchSessions.stream().map(BranchSession::getBranchId).collect(Collectors.toList());
+        List<Long> branchIds = new ArrayList<>(branchSessions.size());
+        boolean notNeedReleaseLock = true;
```

Comment:
already ensure  SessionHolder.findGlobalSession(xid, false) are not in the case.